Gems/AtomLyIntegration/AtomViewportDisplayIcons/Code/Source/AtomViewportDisplayIconsSystemComponent.h

```diff
@@ -79,6 +82,26 @@ namespace AZ
             IconId m_currentId = 0;
 
             bool m_drawContextRegistered = false;
+
+            AZStd::unordered_map<IconId, AZStd::vector<DrawParameters>> m_drawRequests;
+            AzFramework::ViewportId m_drawRequestViewportId = AzFramework::InvalidViewportId;
+            AZ::Data::Instance<AZ::RPI::ShaderResourceGroup> CreateIconSRG(AzFramework::ViewportId viewportId, AZ::Data::Instance<AZ::RPI::Image> image);
+            RHI::Ptr<RPI::DynamicDrawContext> GetDynamicDrawContextForViewport(AzFramework::ViewportId viewportId);
+            AZ::Data::Instance<AZ::RPI::Image> GetImageForIconId(IconId iconId);
+            
+            using Index = AZ::u16;
+            struct Vertex
```

Comment:
Not a fan of generic name like Vertex. Makes code search much harder. Can we use something like IconVertexData?